system/jhtree/jhtree.cpp

```diff
@@ -1539,6 +1573,36 @@ offset_t CKeyIndex::queryMetadataHead()
     return ret;
 }
 
+void CKeyIndex::loadBloomFilter()
+{
+    offset_t bloomAddr = keyHdr->getHdrStruct()->bloomHead;
+    if (!bloomAddr || bloomAddr == static_cast<offset_t>(-1))
+        return; // indexes created before introduction of bloomfilter would have FFFF... in this space
+    uint32_t bloomTableSize = keyHdr->getHdrStruct()->bloomTableSize;
+    unsigned short bloomTableHashes = keyHdr->getHdrStruct()->bloomTableHashes;
+    Owned<CJHTreeBloomTableNode> node;
+    MemoryBuffer bloomTable;
+    bloomTable.ensureCapacity(bloomTableSize);
+    while (bloomAddr)
+    {
+        node.setown(QUERYINTERFACE(loadNode(bloomAddr), CJHTreeBloomTableNode));
```

Comment:
static cast would probably be better, possibly following an assert that the type matches what you expect.